blocks/bricks/recurrent.py

```diff
@@ -300,6 +301,109 @@ def apply(self, input_=None, state=None, mask=None):
         return next_state
 
 
+class LSTM(BaseRecurrent, Initializable):
+    u"""Long Short Term Memory.
+
+    Every unit of an LSTM is equipped with input, forget and output gates.
+    This implementation aims to do as many computations in parallel as
+    possible and expects the last dimension of the input to be four times
+    the output dimension.
+
+    Parameters
+    ----------
+    dim : int
+        The dimension of the hidden state.
+
+    Notes
+    -----
+    See :class:`.Initializable` for initialization parameters.
```

Comment:
I just double checked the computations to see how specific this variant is and apparently it uses peephole connections from the cells to the gates. Somehow the output gates get the current cell values while the other gates receive the cell values from the previous timestep. This is how the LSTM is defined in the paper by Graves but I'm not sure how to name this variant. It's the same architecture that he used for speech recognition as well. Anyway, I saw there is a mistake in the implementation because some of the matrices involved should be diagonal (probably best implemented using element-wise vector multiplications) so I should fix that first before this gets merged. I'm considering to cite Hochreiter et al. as well because that paper coined the name of the architecture.